src/app/core/utils/api-token/api-token.service.ts

```diff
@@ -72,32 +83,35 @@ export class ApiTokenService {
               return { apiToken, type: 'user', isAnonymous: true, creator: 'pwa' };
             } else if (orderId) {
               return { apiToken, type: 'order', orderId, creator: 'pwa' };
-            } else {
-              const apiTokenCookieString = this.cookiesService.get('apiToken');
-              const apiTokenCookie: ApiTokenCookie = apiTokenCookieString
-                ? JSON.parse(apiTokenCookieString)
-                : undefined;
-              if (apiToken && apiTokenCookie) {
+            }
+            const apiTokenCookieString = this.cookiesService.get('apiToken');
+            const apiTokenCookie: ApiTokenCookie = apiTokenCookieString ? JSON.parse(apiTokenCookieString) : undefined;
+            if (apiToken) {
+              if (apiTokenCookie) {
                 return { ...apiTokenCookie, apiToken };
               }
+              return { apiToken, type: 'anonymous', creator: 'pwa', isAnonymous: true };
             }
           }),
-          distinctUntilChanged<ApiTokenCookie>(isEqual)
+          distinctUntilChanged<ApiTokenCookie>(isEqual),
+          withLatestFrom(this.initialCookie$)
         )
-        .subscribe(apiToken => {
+        .subscribe(([apiToken, initialApiToken]) => {
           const cookieContent = apiToken?.apiToken ? JSON.stringify(apiToken) : undefined;
           if (cookieContent) {
-            cookiesService.put('apiToken', cookieContent, {
-              expires: new Date(Date.now() + 3600000),
-              secure: true,
-              sameSite: 'Strict',
-            });
+            if (!isEqual(apiToken, initialApiToken)) {
+              cookiesService.put('apiToken', cookieContent, {
+                expires: apiToken.options?.expires ?? new Date(Date.now() + 3600000),
```

Comment:
Should the expiry date be configurable somehow? Maybe via an injection token or a static/normal property?
It would be easier to change if it weren't hard-coded here.